log/logger.go

```diff
@@ -102,18 +102,20 @@ func (l zeroLogWrapper) Impl() interface{} {
 }
 
 // FilterKeys returns a new logger that filters out all key/value pairs that do not match the filter.
-// This functions assumes that the logger is a zerolog.Logger, which is the case for the logger returned by log.NewLogger()
-// NOTE: filtering has a performance impact on the logger
-func FilterKeys(logger Logger, filter func(key, level string) bool) Logger {
+// This functions assumes that the logger is a zerolog.Logger, which is the case for the logger returned by log.NewLogger().
+// NOTE: filtering has a performance impact on the logger.
+func FilterKeys(logger Logger, filter FilterFunc) Logger {
 	zl, ok := logger.Impl().(*zerolog.Logger)
 	if !ok {
 		panic("logger is not a zerolog.Logger")
 	}
 
 	filteredLogger := zl.Hook(zerolog.HookFunc(func(e *zerolog.Event, lvl zerolog.Level, _ string) {
-		// TODO wait for https://github.com/rs/zerolog/pull/527 to be merged
-		// keys := e.GetKeys()
-		keys := []string{}
+		keys, err := e.GetKeys()
```

Comment:
should we make this method a function that takes keys until the pr is merged? Thinking how we can get this merged

Reply:
Unfortunately, this isn't possible. We have no way to access the log keys from the hook.
I can re-comment this line, and when the zerolog PR is merged we can uncomment it.
That works too for me.

Reply:
awesome lets do it